File: apis/src/main/java/com/beat/apis/performance/application/dto/modify/PerformanceModifyRequest.java
package com.beat.apis.performance.application.dto.modify;

import java.util.List;

import com.beat.apis.performance.application.dto.modify.cast.CastModifyRequest;
import com.beat.apis.performance.application.dto.modify.performanceImage.PerformanceImageModifyRequest;
import com.beat.apis.performance.application.dto.modify.schedule.ScheduleModifyRequest;
import com.beat.apis.performance.application.dto.modify.staff.StaffModifyRequest;
import com.beat.apis.performance.application.dto.BankNameType;
import com.beat.apis.performance.application.dto.GenreType;

import jakarta.validation.constraints.Size;

public record PerformanceModifyRequest(
	Long performanceId,
	String performanceTitle,
	GenreType genre,
	int runningTime,
	@Size(max = 1500, message = "공연 소개는 1500자를 초과할 수 없습니다.")
	String performanceDescription,
	@Size(max = 1500, message = "공연 유의사항은 1500자를 초과할 수 없습니다.")
	String performanceAttentionNote,
	BankNameType bankName,
	String accountNumber,
	String accountHolder,
	String posterImage,
	String performanceTeamName,
	String performanceVenue,
	String roadAddressName,
	String placeDetailAddress,
	String latitude,
	String longitude,
	String performanceContact,
	String performancePeriod,
	int totalScheduleCount,
	int ticketPrice,
	List<ScheduleModifyRequest> scheduleModifyRequests,
	List<CastModifyRequest> castModifyRequests,
	List<StaffModifyRequest> staffModifyRequests,
	List<PerformanceImageModifyRequest> performanceImageModifyRequests
) {
}

File: apis/src/main/kotlin/com/beat/apis/performance/api/request/PerformanceModifyRequest.kt
package com.beat.apis.performance.api.request

import com.beat.apis.performance.api.type.BankNameType
import com.beat.apis.performance.api.type.GenreType
import jakarta.validation.Valid
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Size

data class PerformanceModifyRequest(
    val performanceId: Long,
    val performanceTitle: String,
    val genre: GenreType,
    val runningTime: Int,
    @field:Size(max = 1500, message = "공연 소개는 1500자를 초과할 수 없습니다.")
    val performanceDescription: String,
    @field:Size(max = 1500, message = "공연 유의사항은 1500자를 초과할 수 없습니다.")
    val performanceAttentionNote: String,
    val bankName: BankNameType?,
    val accountNumber: String?,
    val accountHolder: String?,
    val posterImage: String,
    val performanceTeamName: String,
    val performanceVenue: String,
    val roadAddressName: String,
    val placeDetailAddress: String,
    val latitude: String,
    val longitude: String,
    val performanceContact: String,
    @field:NotNull val performancePeriod: String?,
    @field:NotNull val totalScheduleCount: Int?,
    val ticketPrice: Int,
    @field:Valid val scheduleModifyRequests: List<@Valid ScheduleModifyRequest>,
    @field:Valid val castModifyRequests: List<@Valid CastModifyRequest>,
    @field:Valid val staffModifyRequests: List<@Valid StaffModifyRequest>,
    @field:Valid val performanceImageModifyRequests: List<@Valid PerformanceImageModifyRequest>,
)
